packages/vats/src/orchestration.js

```diff
@@ -0,0 +1,268 @@
+  createAccount: M.callWhen(M.string(), M.string())
+    .optional(M.remotable('Port'))
+    .returns(M.remotable('ChainAccount')),
+  getChain: M.callWhen(M.string()).returns(M.remotable('Chain')),
+});
+
+/**
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {ReturnType<typeof prepareChainAccount>} createChainAccount
+ */
+const prepareOrchestration = (zone, createChainAccount) =>
+  zone.exoClassKit(
+    'Orchestration',
+    {
+      admin: M.interface('OrchestrationAdmin', {
+        placeholder: M.call().returns(M.string()),
+      }),
+      public: OrchestrationI,
+    },
+    /** @param {Partial<OrchestrationPowers>} [initialPowers] */
+    initialPowers => {
+      /** @type {PowerStore} */
+      const powers = zone.detached().mapStore('PowerStore');
+      if (initialPowers) {
+        for (const [name, power] of Object.entries(initialPowers)) {
+          powers.init(/** @type {keyof OrchestrationPowers} */ (name), power);
+        }
+      }
+      return { powers, icaControllerNonce: 0 };
+    },
+    {
+      admin: {
+        placeholder() {
+          return 'test';
+        },
+      },
+      public: {
+        /**
+         * @param {import('@agoric/orchestration').ConnectionId} hostConnectionId
+         *   the counterparty connection_id
+         * @param {import('@agoric/orchestration').ConnectionId} controllerConnectionId
+         *   self connection_id
+         * @param {Port} [currentPort] if a port is provided, it will be reused
+         *   to create the account
+         * @returns {Promise<ChainAccount>}
+         */
+        async createAccount(
+          hostConnectionId,
+          controllerConnectionId,
+          currentPort,
+        ) {
+          await null;
+
+          let port;
+          if (!matches(currentPort, M.remotable('Port'))) {
+            if (currentPort) trace('Invalid Port provided, binding a new one.');
+            const network = getPower(this.state.powers, 'network');
+            port = await E(network)
+              .bind(`/ibc-port/icacontroller-${this.state.icaControllerNonce}`)
+              .catch(e => Fail`Failed to get remoteAddress: ${e}`);
+
+            this.state.icaControllerNonce += 1;
+          } else {
+            port = currentPort;
+          }
+          const remoteConnAddr = makeICAConnectionAddress(
+            hostConnectionId,
+            controllerConnectionId,
+          );
+          const chainAccount = createChainAccount(port, remoteConnAddr);
+
+          // await so we do not return a ChainAccount before it successfully instantiates
+          await E(port)
+            .connect(remoteConnAddr, chainAccount.connectionHandler)
+            // XXX if we fail, should we close the port (if it was created in this flow)?
+            .catch(e => Fail`Failed to create ICA connection: ${bare(e)}`);
+
+          return chainAccount.account;
+        },
+        /** @param {string} _chainName e.g. cosmos */
+        getChain(_chainName) {
```

Comment:
I suspect this is at the wrong level of abstraction. I expect we'll have a utility in front of this vat which manages chain config.

Reply:
Acknowledged. I think the idea is to be able to do: 
```js
const cosmos = E(orchestration).getChain('cosmos');
const account = E(cosmos).createAccount();
```

And `getChain` might look something like: 

```js
getChain(chainName) {
 return E(agoricNames || someOtherNamehub).lookup('chain', chainName); 
}
```

Nonetheless, out of scope for this PR (_in scope for #9063_) and was intended just to be a placeholder based on this [proposed spec](https://github.com/agoric-labs/orchestration-api-spec/blob/dfbf6f09d46bb8e0fdc4b16125e0c90f899b2b58/src/index.d.ts).  Happy to remove from this PR if you'd prefer

Reply:
yes, please remove the placeholders. ideally everything includes is tested